packages/platforms/react-native/ios/BugsnagReactNativePerformance.mm

```diff
@@ -216,35 +231,40 @@ static uint64_t hexStringToUInt64(NSString *hexString) {
                 resolve:(RCTPromiseResolveBlock)resolve
                 reject:(RCTPromiseRejectBlock)reject) {
     NSString *spanKey = [spanId stringByAppendingString:traceId];
-    BugsnagPerformanceSpan *nativeSpan = openSpans[spanKey];
-    if (nativeSpan != nil) {
-        [openSpans removeObjectForKey:spanKey];
-        
-        // Set native span attributes from JS values
-        [ReactNativeSpanAttributes setNativeAttributes:nativeSpan.attributes fromJSAttributes:attributes];
 
-        // We need to reinstate the bugsnag.sampling.p attribute here as it might not be re-populated on span end
-        nativeSpan.attributes[@"bugsnag.sampling.p"] = @(nativeSpan.samplingProbability);
-        
-        // If the end time is later than the current end time, update it
-        NSDate *nativeEndTime = [NSDate dateWithTimeIntervalSince1970: endTime / NSEC_PER_SEC];
-        if ([nativeEndTime timeIntervalSinceDate:nativeSpan.endTime] > 0) {
-            [nativeSpan markEndTime:nativeEndTime];
+    @synchronized (openSpans) {
+        BugsnagPerformanceSpan *nativeSpan = openSpans[spanKey];
+        if (nativeSpan != nil) {
+            [openSpans removeObjectForKey:spanKey];
+            
+            // Set native span attributes from JS values
+            [ReactNativeSpanAttributes setNativeAttributes:nativeSpan.attributes fromJSAttributes:attributes];
+
+            // We need to reinstate the bugsnag.sampling.p attribute here as it might not be re-populated on span end
+            nativeSpan.attributes[@"bugsnag.sampling.p"] = @(nativeSpan.samplingProbability);
+            
+            // If the end time is later than the current end time, update it
+            NSDate *nativeEndTime = [NSDate dateWithTimeIntervalSince1970: endTime / NSEC_PER_SEC];
+            if ([nativeEndTime timeIntervalSinceDate:nativeSpan.endTime] > 0) {
+                [nativeSpan markEndTime:nativeEndTime];
+            }
+            
+            [nativeSpan sendForProcessing];
```

Comment:
I think that these parts don't need to be inside of the synchronized code block?

Reply:
ah yeah, good point 👍🏾